common/src/main/java/com/graphicsfuzz/common/util/PipelineUniformValueSupplier.java

```diff
@@ -30,6 +30,10 @@ public PipelineUniformValueSupplier(PipelineInfo pipelineInfo) {
 
   @Override
   public Optional<List<String>> getValues(String name) {
-    return Optional.of(this.pipelineInfo.getArgs(name));
+    if (pipelineInfo.hasUniform(name)) {
+      return Optional.of(this.pipelineInfo.getArgs(name));
+    } else {
+      return Optional.empty();
+    }
```

Comment:
It's a matter of taste really, but the "else" is redundant; you can always rewrite:

```
if (e) {
  return f;
} else {
  return g;
}
```

as

```
if (e) {
  return f;
}
return g;
```

I slightly prefer the latter.

Reply:
Rewritten.